src/it/resources/com/google/checkstyle/test/chapter5naming/rule522typenames/InputClassNames.java

```diff
@@ -0,0 +1,83 @@
+package com.google.checkstyle.test.chapter5naming.rule522typenames;
+
+class inputHeaderClass {
+    // violation above 'Type name 'inputHeaderClass' must match pattern .*'
+
+    public interface InputHeader___Interface {};
+    // violation above 'Type name 'InputHeader___Interface' .* .*'
+
+    public enum inputHeaderEnum { one, two };
+    // violation above 'Type name 'inputHeaderEnum' must match pattern .*'
+
+    class NoValid$Name {}
+    // violation above 'Type name '.*' must match pattern .*'
```

Comment:
Remove trailing .* It should not be required

Reply:
done